### ISSUE TITLE
how can keep scroll position when unshift items to top?

### ISSUE THREAD

**Issue by Treri** [tangbc/vue-virtual-scroll-list#28]
How can keep scroll position when unshift item in top?

Content starts like:
```
content top:     -----------
                 B
viewport top:    ===========
                 C
                 D
viewport bottom: ===========
```

And then a new item is added to the top (A):

```
content top:     -----------
                 A
viewport top:    ===========
                 B
                 C
viewport bottom: ===========
                 D
```

Instead of the user continuing to see C at the top of the viewport, C and D have been jarringly pushed down due to the way content typically flows. What we want to happen is:
```
content top:     -----------
                 A
                 B
viewport top:    ===========
                 C
                 D
viewport bottom: ===========
```

**Comment by tangbc** [comment]
Er, this case is better to handled by youself.

I've come up with a solution: first, store the virtual list viewport top index as `currentIndex` at `onscroll` function, and when you unshift a item in top, you set the `start` prop as `currentIndex + 1` immediately.

The `currentIndex` is the viewport top index, it can be work out likely by this way:
```javascript
// assign the `onscroll` prop
onscroll (evt, data) {
    let size = this.size // the prop `size` value
    this.currentIndex = Math.floor(data.offset / size)
},
unshiftItem: function () {
    this.items.unshift('xxx')
    // this.start is virtual list's prop `start`
    this.start = this.currentIndex + 1
}
```

This may be solving you problem, you can try it.

**Comment by Treri** [comment]
Thanks very much.
This can solve the problem, though not perfect.

The topIndex is a integer, when I unshift items, the scroll position is may not the top of topIndex. Then user feel the content jumps to the boundary of the the first item before unshift

**Comment by tangbc** [comment]
Yes, you've remind me think of adding a new prop `offset` which is same to prop `start`, but `offset` is using for setting a custom scroll offset.

I will add this feature support in next version today or tomorrow.

**Comment by tangbc** [comment]
This feature is released in [v1.2.0](https://github.com/tangbc/vue-virtual-scroll-list/releases/tag/v1.2.0)
